source/material_model/melt_boukare.cc

```diff
@@ -0,0 +1,1260 @@
+                      else if (c == porosity_idx && this->get_timestep_number() > 0)
+                        reaction_rate_out->reaction_rates[q][c] = porosity_change / reaction_time_step_size;
+                      else
+                        reaction_rate_out->reaction_rates[q][c] = 0.0;
+                    }
+                  out.reaction_terms[q][c] = 0.0;
+                }
+
+              if (enthalpy_out != nullptr)
+                {
+                  const double melt_molar_mass = endmember_mole_fractions_per_phase[mgmelt_idx] * molar_masses[mgmelt_idx]
+                                                 + endmember_mole_fractions_per_phase[femelt_idx] * molar_masses[femelt_idx];
+                  const double Fe_enthalpy_of_fusion = Fe_mantle_melting_temperature * Fe_mantle_melting_entropy
+                                                       + (this->get_adiabatic_conditions().pressure(in.position[q]) - melting_reference_pressure) * Fe_mantle_melting_volume;
+                  const double Mg_enthalpy_of_fusion = Mg_mantle_melting_temperature * Mg_mantle_melting_entropy
+                                                       + (this->get_adiabatic_conditions().pressure(in.position[q]) - melting_reference_pressure) * Mg_mantle_melting_volume;
+                  double enthalpy_of_fusion = Fe_enthalpy_of_fusion * bulk_composition + Mg_enthalpy_of_fusion * (1.0-bulk_composition);
+                  enthalpy_of_fusion /= melt_molar_mass;
+
+                  enthalpy_out->enthalpies_of_fusion[q] = enthalpy_of_fusion;
+                }
+
+              // cutoff for viscosity at 30%
+              const double porosity = std::min(0.3, std::max(in.composition[q][porosity_idx],0.0));
+              out.viscosities[q] = (1.0 - porosity) * eta_0 * exp(- alpha_phi * porosity);
+            }
+          else
+            {
+              out.viscosities[q] = eta_0;
+
+              // no melting/freezing is used in the model --> set all reactions to zero
+              for (unsigned int c=0; c<in.composition[q].size(); ++c)
+                {
+                  if (reaction_rate_out != nullptr)
+                    reaction_rate_out->reaction_rates[q][c] = 0.0;
+                }
+            }
+
+          out.entropy_derivative_pressure[q]    = 0.0;
+          out.entropy_derivative_temperature[q] = 0.0;
+          out.thermal_conductivities[q] = thermal_conductivity;
+
+          for (unsigned int c=0; c<in.composition[q].size(); ++c)
+            out.reaction_terms[q][c] = 0.0;
+
+          double visc_temperature_dependence = 1.0;
+          const double delta_temp = in.temperature[q]-this->get_adiabatic_conditions().temperature(in.position[q]);
+          visc_temperature_dependence = std::max(std::min(std::exp(-thermal_viscosity_exponent*delta_temp/this->get_adiabatic_conditions().temperature(in.position[q])),1e8),1e-8);
```

Comment:
and delete line 858 above
```suggestion
          const double visc_temperature_dependence = std::max(std::min(std::exp(-thermal_viscosity_exponent*delta_temp/this->get_adiabatic_conditions().temperature(in.position[q])),1e8),1e-8);
```